### ISSUE TITLE
error: Run modelThe model and loaded state dict do not match exactly

### ISSUE THREAD

**Issue by 363546178** [hustvl/MapTR#22]
Hi! When I run MapTR-tiny model, it prompts the following error:

```

load checkpoint from local path: ./ckpts/maptr_tiny_r50_24e.pth
The model and loaded state dict do not match exactly

unexpected key in source state_dict: pts_bbox_head.transformer.encoder.layers.0.attentions.1.attention.grid_offsets


```
The executed command is as follows:

```
/tools/dist_test_map.sh ./projects/configs/maptr/maptr_tiny_r50_24e.py ./ckpts/maptr_tiny_r50_24e.pth 1 
```

Could this be a problem caused by a parameter input error? If yes is there a good solution? Thanks!

Appendix: All LOG information output during the model run

```
(maptr) ➜  MapTR git:(main) ✗ ./tools/dist_test_map.sh ./projects/configs/maptr/maptr_tiny_r50_24e.py ./ckpts/maptr_tiny_r50_24e.pth 1 
/home/test/anaconda3/envs/maptr/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torch.distributed.run.
Note that --use_env is set by default in torch.distributed.run.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
projects.mmdet3d_plugin
/home/test/code/MapTR/projects/mmdet3d_plugin/bevformer/modules/custom_base_transformer_layer.py:94: UserWarning: The arguments `feedforward_channels` in BaseTransformerLayer has been deprecated, now you should set `feedforward_channels` and other FFN related arguments to a dict named `ffn_cfgs`. 
  warnings.warn(
/home/test/code/MapTR/projects/mmdet3d_plugin/bevformer/modules/custom_base_transformer_layer.py:94: UserWarning: The arguments `ffn_dropout` in BaseTransformerLayer has been deprecated, now you should set `ffn_drop` and other FFN related arguments to a dict named `ffn_cfgs`. 
  warnings.warn(
/home/test/code/MapTR/projects/mmdet3d_plugin/bevformer/modules/custom_base_transformer_layer.py:94: UserWarning: The arguments `ffn_num_fcs` in BaseTransformerLayer has been deprecated, now you should set `num_fcs` and other FFN related arguments to a dict named `ffn_cfgs`. 
  warnings.warn(
/home/test/anaconda3/envs/maptr/lib/python3.8/site-packages/mmcv/cnn/bricks/transformer.py:341: UserWarning: The arguments `feedforward_channels` in BaseTransformerLayer has been deprecated, now you should set `feedforward_channels` and other FFN related arguments to a dict named `ffn_cfgs`. 
  warnings.warn(
/home/test/anaconda3/envs/maptr/lib/python3.8/site-packages/mmcv/cnn/bricks/transformer.py:341: UserWarning: The arguments `ffn_dropout` in BaseTransformerLayer has been deprecated, now you should set `ffn_drop` and other FFN related arguments to a dict named `ffn_cfgs`. 
  warnings.warn(
/home/test/anaconda3/envs/maptr/lib/python3.8/site-packages/mmcv/cnn/bricks/transformer.py:341: UserWarning: The arguments `ffn_num_fcs` in BaseTransformerLayer has been deprecated, now you should set `num_fcs` and other FFN related arguments to a dict named `ffn_cfgs`. 
  warnings.warn(
/home/test/anaconda3/envs/maptr/lib/python3.8/site-packages/mmcv/cnn/bricks/transformer.py:92: UserWarning: The arguments `dropout` in MultiheadAttention has been deprecated, now you can separately set `attn_drop`(float), proj_drop(float), and `dropout_layer`(dict) 
  warnings.warn('The arguments `dropout` in MultiheadAttention '
load checkpoint from local path: ./ckpts/maptr_tiny_r50_24e.pth
The model and loaded state dict do not match exactly

unexpected key in source state_dict: pts_bbox_head.transformer.encoder.layers.0.attentions.1.attention.grid_offsets

[                                                  ] 0/6019, elapsed: 0s, ETA:/home/test/anaconda3/envs/maptr/lib/python3.8/site-packages/torch/_tensor.py:575: UserWarning: floor_divide is deprecated, and will be removed in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values.
To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor'). (Triggered internally at  ../aten/src/ATen/native/BinaryOps.cpp:467.)
  return torch.floor_divide(self, other)
[                                                  ] 1/6019, 0.3 task/s, elapsed: 4s, ETA: 22939s/home/test/code/MapTR/projects/mmdet3d_plugin/core/bbox/coders/nms_free_coder.py:236: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  self.post_center_range = torch.tensor(
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6019/6019, 1.6 task/s, elapsed: 3695s, ETA:     0s
Formating bboxes of pts_bbox
Start to convert map detection format...
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6019/6019, 53846.5 task/s, elapsed: 0s, ETA:     0s
data/nuscenes/nuscenes_map_anns_val.json exist, not update
Results writes to test/maptr_tiny_r50_24e/Sat_Feb__4_12_41_21_2023/pts_bbox/nuscmap_results.json
Evaluating bboxes of pts_bbox
Formating results & gts by classes
results path: /home/test/code/MapTR/test/maptr_tiny_r50_24e/Sat_Feb__4_12_41_21_2023/pts_bbox/nuscmap_results.json
Formatting ...
Cls data formatting done in 21.579294s!! with /home/test/code/MapTR/test/maptr_tiny_r50_24e/Sat_Feb__4_12_41_21_2023/pts_bbox/cls_formatted.pkl
-*-*-*-*-*-*-*-*-*-*use metric:chamfer-*-*-*-*-*-*-*-*-*-*
-*-*-*-*-*-*-*-*-*-*threshhold:0.5-*-*-*-*-*-*-*-*-*-*
cls:divider done in 1.090212s!!
cls:ped_crossing done in 0.051536s!!
cls:boundary done in 0.068384s!!

+--------------+-------+------+--------+-------+
| class        | gts   | dets | recall | ap    |
+--------------+-------+------+--------+-------+
| divider      | 27332 | 0    | 0.000  | 0.000 |
| ped_crossing | 6406  | 0    | 0.000  | 0.000 |
| boundary     | 21050 | 0    | 0.000  | 0.000 |
+--------------+-------+------+--------+-------+
| mAP          |       |      |        | 0.000 |
+--------------+-------+------+--------+-------+
-*-*-*-*-*-*-*-*-*-*threshhold:1.0-*-*-*-*-*-*-*-*-*-*
cls:divider done in 1.094373s!!
cls:ped_crossing done in 0.058753s!!
cls:boundary done in 0.068401s!!

+--------------+-------+------+--------+-------+
| class        | gts   | dets | recall | ap    |
+--------------+-------+------+--------+-------+
| divider      | 27332 | 0    | 0.000  | 0.000 |
| ped_crossing | 6406  | 0    | 0.000  | 0.000 |
| boundary     | 21050 | 0    | 0.000  | 0.000 |
+--------------+-------+------+--------+-------+
| mAP          |       |      |        | 0.000 |
+--------------+-------+------+--------+-------+
-*-*-*-*-*-*-*-*-*-*threshhold:1.5-*-*-*-*-*-*-*-*-*-*
cls:divider done in 1.080050s!!
cls:ped_crossing done in 0.055158s!!
cls:boundary done in 0.067125s!!

+--------------+-------+------+--------+-------+
| class        | gts   | dets | recall | ap    |
+--------------+-------+------+--------+-------+
| divider      | 27332 | 0    | 0.000  | 0.000 |
| ped_crossing | 6406  | 0    | 0.000  | 0.000 |
| boundary     | 21050 | 0    | 0.000  | 0.000 |
+--------------+-------+------+--------+-------+
| mAP          |       |      |        | 0.000 |
+--------------+-------+------+--------+-------+
divider: 0.0
ped_crossing: 0.0
boundary: 0.0
map: 0.0
{'NuscMap_chamfer/divider_AP': 0.0, 'NuscMap_chamfer/ped_crossing_AP': 0.0, 'NuscMap_chamfer/boundary_AP': 0.0, 'NuscMap_chamfer/mAP': 0.0, 'NuscMap_chamfer/divider_AP_thr_0.5': 0.0, 'NuscMap_chamfer/divider_AP_thr_1.0': 0.0, 'NuscMap_chamfer/divider_AP_thr_1.5': 0.0, 'NuscMap_chamfer/ped_crossing_AP_thr_0.5': 0.0, 'NuscMap_chamfer/ped_crossing_AP_thr_1.0': 0.0, 'NuscMap_chamfer/ped_crossing_AP_thr_1.5': 0.0, 'NuscMap_chamfer/boundary_AP_thr_0.5': 0.0, 'NuscMap_chamfer/boundary_AP_thr_1.0': 0.0, 'NuscMap_chamfer/boundary_AP_thr_1.5': 0.0}

```

**Comment by LegendBC** [comment]
The reporting warning will not influence the metric. Can you provide the version of your shapely?

**Comment by ZZZENG** [comment]
> > The reporting warning will not influence the metric. Can you provide the version of your shapely?
> 
> I have the same problem, the version of my shapely is 1.8.5.post1.

I find the reason, we should use the models provided in README.md instead of the pretrained models provided in install.md.

**Comment by 363546178** [comment]
I changed the computer video card，the old video card is Nvidia GTX1660S，new video card is Nvidia RTX2060，this problem is sloved in RTX2060,the code and model not changed. I think the old video card (GTX1660S) have some troules  caused the models can't work normally.😷
